main.go

```diff
@@ -1014,58 +1023,152 @@ func processImportCfg(flags []string) (newImportCfg string, _ error) {
+var knownReflectAPIs = map[funcFullName][]reflectArgPosition{
+	"reflect.TypeOf":  {0},
+	"reflect.ValueOf": {0},
+}
+
+func loadKnownReflectAPIs() (map[string][]int, error) {
+	dir := filepath.Join(os.TempDir(), "garble")
+	filename := filepath.Join(dir, "reflect.json")
+
+	data, err := os.ReadFile(filename)
+	if err != nil || len(data) == 0 {
+		return knownReflectAPIs, nil
+	}
+
+	var newKnownReflectAPIs map[string][]int
+
+	err = json.Unmarshal(data, &newKnownReflectAPIs)
+
+	if len(newKnownReflectAPIs) == 0 {
+		return knownReflectAPIs, err
+	}
+
+	return newKnownReflectAPIs, err
+}
+
+func saveKnownReflectAPIs() error {
+	dir := filepath.Join(os.TempDir(), "garble")
+	filename := filepath.Join(dir, "reflect.json")
+	lockfileName := filename + ".lck"
+
+	if err := os.MkdirAll(dir, 0777); err != nil {
+		return err
+	}
+
+	lck, err := lockfile.New(lockfileName)
+	if err != nil {
+		return err
+	}
+
+	// loop until we get the lock
+	for err := lck.TryLock(); err != nil; err = lck.TryLock() {
+		time.Sleep(1 * time.Millisecond)
+	}
+
+	defer lck.Unlock()
+
+	curr, err := loadKnownReflectAPIs()
+	if err != nil {
+		return err
+	}
+
+	for k, v := range curr {
+		knownReflectAPIs[k] = v
+	}
+
+	data, err := json.Marshal(knownReflectAPIs)
+	if err != nil {
+		return err
+	}
+
+	return os.WriteFile(filename, data, 0666)
+}
+
+func (tf *transformer) findReflectFunctions(files []*ast.File) {
+	visitReflect := func(node ast.Node) {
+		funcDecl, ok := node.(*ast.FuncDecl)
+		if !ok {
+			return
+		}
+
+		funcObj := tf.info.ObjectOf(funcDecl.Name).(*types.Func)
+
+		var paramNames []string
+		for _, param := range funcDecl.Type.Params.List {
+			for _, name := range param.Names {
+				paramNames = append(paramNames, name.Name)
+			}
+		}
+
+		ast.Inspect(funcDecl, func(node ast.Node) bool {
+			call, ok := node.(*ast.CallExpr)
+			if !ok {
+				return true
+			}
+			sel, ok := call.Fun.(*ast.SelectorExpr)
+			if !ok {
+				return true
+			}
+			fnType, _ := tf.info.ObjectOf(sel.Sel).(*types.Func)
+			if fnType == nil || fnType.Pkg() == nil {
+				return true
+			}
+
+			fullName := fnType.FullName()
+			var identifiers []string
+			for _, argPos := range knownReflectAPIs[fullName] {
+				arg := call.Args[argPos]
+
+				ident, ok := arg.(*ast.Ident)
+				if !ok {
+					continue
+				}
+
+				obj := tf.info.ObjectOf(ident)
+				if obj.Parent() == funcObj.Scope() {
+					identifiers = append(identifiers, ident.Name)
+				}
+			}
+
+			if identifiers == nil {
+				return true
+			}
+
+			var argumentPosReflect []int
+			for _, ident := range identifiers {
+				for paramPos, paramName := range paramNames {
+					if ident == paramName {
+						argumentPosReflect = append(argumentPosReflect, paramPos)
+					}
+				}
+			}
+			knownReflectAPIs[funcObj.FullName()] = argumentPosReflect
+
+			return true
+		})
+	}
+
+	lenPrevKnownReflectAPIs := len(knownReflectAPIs)
+	for _, file := range files {
+		for _, decl := range file.Decls {
+			visitReflect(decl)
+		}
+	}
```

Comment:
you've got a better route than looping over files+decls, walking, filtering FuncDecls, and calling ObjectOf.

you can replace all of that with something like:

```
pkgScope := tf.pkg.Scope()
for _, name := range pkgScope.Names() {
    funcObj, ok := pkgScope.Lookup(name).(*types.Func)
    if !ok {
        continue
    }
    // ...
```

Reply:
Then again, you do need to look at the func's AST, so you probably still need to ast.Inspect first. I'll leave this here in case it's still helpful.